src/ovirtapi/transform.js

```diff
@@ -202,8 +204,15 @@ const VM = {
     }
 
     if (vm.nics && vm.nics.nic) {
+      const reportedDevicesArray =
+        (vm.reported_devices && vm.reported_devices.reported_device)
+          ? vm.reported_devices.reported_device.filter(device => device.type === 'network')
+          : []
+
+      const reportedDeviceMap = arrayToMap(reportedDevicesArray, device => device.mac.address)
+
       parsedVm.nics = vm.nics.nic.map(
-        nic => Nic.toInternal({ nic })
+        nic => Nic.toInternal({ nic, reportedDevice: reportedDeviceMap[nic.mac.address] })
```

Comment:
We assume here that `nic.mac` can be matched with `reportedDevice.mac` however this seems not to be true all the time (tested on brq by adding `?follow=reporteddevices` to `OvirtApi.getVmNic()`) i.e. for query [1] the device returned by backend has has the same MAC (not matching any NIC) for all reported cards. 
Note also that `mac` element is optional (see [ReportedDeviceMapper](https://github.com/oVirt/ovirt-engine/blob/07969f6cd0878d39a42d4feab0871e86a22e501b/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/ReportedDeviceMapper.java#L27)) so we need to filter out invalid entities.

We could avoid those problems by adding `?follow=nics.reporteddevices` - this way the backend will do that for us. 

[1] .../api/vms/8f8bb2b8-34c1-4e5a-b4ec-c5301bfc14ac/nics?follow=reporteddevices

Reply:
OK, in the code it can be optional but from what I know each network interface card should have it's own unique MAC address so I'm no sure if it's "over kill" in the code or there is any possible scenario that the mac will be undefined?

> We could avoid those problems by adding ?follow=nics.reporteddevices - this way the backend will do that for us.

In that case it's return all the ips for each nic and than we gather the wrong data, and it's duplicated in the view:
![image](https://user-images.githubusercontent.com/64131213/111463599-6ea9de80-8728-11eb-8ae1-2dea3ea3e43e.png)

The reason  I used the MAC as unique is because I didn't found any relation between the nic and the reported data except MAC address.
but we are definitely can add the `?follow=reporteddevices` to the fetchVmNics call

Reply:
In the case you raised in your comment:
> i.e. for query [1] the device returned by backend has has the same MAC (not matching any NIC) for all reported cards.

The filtering by MAC will filter it, and that's what expected, because the vm is down and it should not have an ip.

Reply:
> we gather the wrong data, and it's duplicated in the view
>

Backend returns duplicated data?

> relation between the nic and the reported data except MAC address
> 
The  relation between devices and nics is based on device UUID - this is basically what  is returned from `.../nics/123?follow=reporteddevices`.

Reply:
> Backend returns duplicated data?

Yes,
first nic:
![image](https://user-images.githubusercontent.com/64131213/111515264-3cfe3b00-875b-11eb-80a5-281984bc8440.png)
second nic:
![image](https://user-images.githubusercontent.com/64131213/111515771-d9c0d880-875b-11eb-9e62-1ac7bb502dc0.png)

You can take a look on my screenshots, or try it on this vm on brq-dev: 95a41852-2898-4b51-8131-6e4315b65668
It's filtered in the backend

Reply:
>> Backend returns duplicated data?
>
> Yes,
>

That's strange.
1. Might be a bug. Then we should report it and let the backend fix it.
2. Might be stale data i.e. old card still visible (but the new one is already in the list) or just delay/synchronization issues when reading the config from underlying hardware.

In both cases I would remove the MAC matching algorithm and treat the backend as the ultimate source of truth. If duplicated addresses look ugly then we can remove duplicates in the React component.(view layer). But let's treat thme just as strings (duplicated strings in a list of strings) - no custom logic.
The proper solution would be probably to display both nick name and nick address (then duplicated addresses would make sense).